system-variables.md

```diff
@@ -1964,6 +1974,14 @@ explain select * from t where age=5;
 - By default, Regions are split for a new table when it is being created in TiDB. After this variable is enabled, the newly split Regions are scattered immediately during the execution of the `CREATE TABLE` statement. This applies to the scenario where data need to be written in batches right after the tables are created in batches, because the newly split Regions can be scattered in TiKV beforehand and do not have to wait to be scheduled by PD. To ensure the continuous stability of writing data in batches, the `CREATE TABLE` statement returns success only after the Regions are successfully scattered. This makes the statement's execution time multiple times longer than that when you disable this variable.
 - Note that if `SHARD_ROW_ID_BITS` and `PRE_SPLIT_REGIONS` have been set when a table is created, the specified number of Regions are evenly split after the table creation.
 
+### `tidb_shard_allocate_step` <span class="version-mark">New in v5.0</span>
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Default value: `9223372036854775807`
+- Range: `[1, 9223372036854775807]`
+- This variable controls the max number of the continous IDs. Generally, the allocating [`AUTO_RANDOM`](/auto-random.md) IDs or the [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md) annotated row IDs are incremental and continuous in one transaction. This variable can be used to solve the hotspot issue in the large transaction scenario.
```

Comment:
> max number of the continous IDs

Should it be 'max number of the continuous IDs in one single SQL statement'?

Reply:
Nope. It should be one transaction:

```sql
mysql> create table t (a bigint auto_random primary key, b int);
Query OK, 0 rows affected, 1 warning (0.04 sec)

mysql> set @@tidb_shard_allocate_step = 3;
Query OK, 0 rows affected (0.00 sec)

mysql> begin;
Query OK, 0 rows affected (0.01 sec)

mysql> insert into t (b) values (1), (2);
Query OK, 2 rows affected (0.00 sec)
Records: 2  Duplicates: 0  Warnings: 0

mysql> insert into t (b) values (3), (4);
Query OK, 2 rows affected (0.00 sec)
Records: 2  Duplicates: 0  Warnings: 0

mysql> select * from t;
+---------------------+------+
| a                   | b    |
+---------------------+------+
| 1441151880758558721 |    1 |
| 1441151880758558722 |    2 |
| 1441151880758558723 |    3 |
| 4899916394579099652 |    4 |
+---------------------+------+
4 rows in set (0.00 sec)
```

Reply:
I see, thanks!